control-plane/p4RuntimeArchStandard.cpp

```diff
@@ -1024,6 +1030,31 @@ class P4RuntimeArchHandlerPSA final : public P4RuntimeArchHandlerCommon<Arch::PS
 
         return Digest{decl->controlPlaneName(), typeSpec, decl->to<IR::IAnnotated>()};
     }
+
+    /// @return true if @table's 'support_timeout' property exists and is true. This
+    /// indicates that @table supports entry ageing.
+    static bool getSupportsTimeout(const IR::P4Table* table) {
+        auto timeout = table->properties->getProperty("psa_idle_timeout");
+
+        if (timeout == nullptr) return false;
+
+        if (auto exprValue = timeout->value->to<IR::ExpressionValue>()) {
```

Comment:
This will only work if the expression is a compile-time known value (i.e., constant).
Is this documented? Perhaps the error message should say that.

Reply:
@mbudiu-vmw, we do have usage mentioned in the psa spec,

https://p4.org/p4-spec/docs/PSA-v1.1.0.html

> 7.2.1. Table entry timeout notification
> PSA uses the psa_idle_timeout to enable a table implementation send notifications from the PSA device when a configurable time has passed since an entry was last matched. The property may take one of two values – NO_TIMEOUT, and NOTIFY_CONTROL. 

In `psa.p4` this is represented as an enum,
https://github.com/p4lang/p4c/blob/84051411d31a39d5670b4721f438fa2f3d232d60/p4include/psa.p4#L263-L267

and will show up as,

```
[268478] ExpressionValue
  expression: [268479] Member member=NOTIFY_CONTROL
    type: [919] Type_Enum name=PSA_IdleTimeout_t declid=36
      annotations: [2] Annotations
      [916] Declaration_ID name=NO_TIMEOUT declid=70
      [917] Declaration_ID name=NOTIFY_CONTROL declid=71
    expr: [268480] TypeNameExpression
      type: [253042] Type_Type
        type: [919] Type_Enum name=PSA_IdleTimeout_t declid=36...
      typeName: [2481] Type_Name
        path: [2480] Path name=PSA_IdleTimeout_t absolute=0
```

I can explicitly add the allowed values in the error message.

Reply:
How about this case: 
```
PSA_IdleTimeout_t v = PSA_IdleTimeout_t.NO_TIMEOUT;
// v could also come from an external function call, or some other dynamic expression.
table t {
   ... 
    psa_idle_timeout = v;
}
```

Reply:
@mbudiu-vmw , your example results in a `IR::PathExpression`

```
[191647] ExpressionValue
  expression: [191648] PathExpression
    type: [919] Type_Enum name=PSA_IdleTimeout_t declid=36
      annotations: [2] Annotations
      [916] Declaration_ID name=NO_TIMEOUT declid=70
      [917] Declaration_ID name=NOTIFY_CONTROL declid=71
    path: [147079] Path name=v_0/v absolute=0

```
Since this is not resolved, I am not sure how to handle it here.

Reply:
For this case you should just give an error requiring the use of a constant value.

Reply:
Added an error message for this case.